httpbakery/error.go

```diff
@@ -38,8 +39,23 @@ type Error struct {
 	Code    ErrorCode  `json:",omitempty"`
 	Message string     `json:",omitempty"`
 	Info    *ErrorInfo `json:",omitempty"`
+
+	// version holds the protocol version that was used
+	// to create the error (see NewDischargeRequiredErrorWithVersion).
+	version version
 }
 
+// version represents a version of the bakery protocol. It is jused
+// to determine the kind of response to send when there is a
+// discharge-required error.
+type version int
+
+const (
+	version0      version = 0
```

Comment:
I went back and forth on this. I ended up thinking they are useful because they enumerate all the versions that we explicitly support.